src/utils/__tests__/analytics.js

```diff
@@ -0,0 +1,95 @@
+import getFilterString, { filterComponents, rangeComponents, rangeObjectComponents, parseRangeObject, parseFilterValue } from '../analytics';
+
+test('snapshot filter components', () => {
+	expect(filterComponents).toMatchSnapshot();
+});
+
+test('snapshot range components', () => {
+	expect(rangeComponents).toMatchSnapshot();
+});
+
+test('snapshot range object components', () => {
+	expect(rangeObjectComponents).toMatchSnapshot();
+});
+
+test('parseRangeObject should parse the range object', () => {
+	const filterKey = 'filter';
+	const rangeObject = {
+		start: 1,
+		end: 3,
+	};
+	const res = parseRangeObject(filterKey, rangeObject);
+	expect(res).toBe('filter=1~3');
+});
+
+test('parseFilterValue should parse string values', () => {
+	const componentId = 'id1';
+	const componentValues = {
+		label: 'Filter',
+		value: 'value1',
+	};
+	const res = parseFilterValue(componentId, componentValues);
+	expect(res).toBe('Filter=value1');
+});
+
+test('parseFilterValue should parse string values and fallback to componentId', () => {
+	const componentId = 'id1';
+	const componentValues = {
+		value: 'value1',
+	};
+	const res = parseFilterValue(componentId, componentValues);
+	expect(res).toBe('id1=value1');
+});
+
+test('parseFilterValue should parse array values', () => {
+	const componentId = 'id1';
+	const componentValues = {
+		label: 'Filter',
+		value: ['value1', 'value2'],
+	};
+	const res = parseFilterValue(componentId, componentValues);
+	expect(res).toBe('Filter=value1,Filter=value2');
+});
+
+test('parseFilterValue should parse array of object values', () => {
+	const componentId = 'id1';
+	const componentValues = {
+		label: 'Filter',
+		value: [{ label: 'label1', value: 'value1' }],
```

Comment:
You should also include the test for multiple values in the array and null values.

Reply:
Added test for multiple values.

The `parseFilterValue` will return `null` as a filter value independently since it just parses the value. But a `null` value shouldn't get through which `getFilterString` is taking care of by filtering out components having `null` or falsy value.

Reply:
cool beans